### PR TITLE
Positioning update

### PR DESCRIPTION
**Pull request by oluoluoxenfree**

### CODE REVIEW

File: src/filters/pages.cjs
Comment (jgerigmeyer) [inline src/filters/pages.cjs:373]: @oluoluoxenfree I think this is the change that was needed to fix this filter, and it's IMO an annoying inconsistency in 11ty. Within the context of a given page, data can be access directly, e.g. `page.home`. But within the context of a _page in a collection_, the data is stored under a named `data` key, e.g. `page.data.home`.

https://www.11ty.dev/docs/collections/#collection-item-data-structure

File: content/index.md
Comment (jgerigmeyer) [inline content/index.md:53]: This is now "working", but is running into an issue similar to what we've found on Sass Site -- because this is a Markdown file, the HTML returned from this macro is then being parsed as Markdown, which we don't want. @mirisuzanne Suggestion for how to fix this? I suppose one option would be upgrading 11ty and then using `renderFile` to include this?

Comment (mirisuzanne) [reply content/index.md:53]: Yeah, upgrading is not a bad idea anyway? The other alternative is making this an NJK file, and using markdown shortcodes where we want to keep markdown parsing.

Comment (jgerigmeyer) [reply content/index.md:53]: Makes sense. @oluoluoxenfree Is this something you'd like me to take a look at, or would you prefer to do it if I send some links/pointers? I'm fine either way.

Comment (oluoluoxenfree) [reply content/index.md:53]: I'm happy for you to do it so I can read your PR/comments to see how you go about it! I'm never sure with stuff like that

Comment (jgerigmeyer) [reply content/index.md:53]: Fixed in #327.

**Comment by jgerigmeyer** [comment]
[Update homepage description based on positioning retreat conversation](https://trello.com/c/spZKQTWm/189-update-homepage-description-based-on-positioning-retreat-conversation)

File: src/scss/layout/_title.scss
Comment (mirisuzanne) [inline src/scss/layout/_title.scss:13]: I'm not sure what this is for, but `50px` feels like a magic number we should clarify, and not obviously related to anything else in our sizing. What's the goal here?

File: src/scss/layout/_title.scss
Comment (mirisuzanne) [inline src/scss/layout/_title.scss:18]: Because of the negative margin above, this is breaking pages that have a hero image:
 
Screenshot 2023-03-08 at 2 04 46 PM

File: src/scss/layout/_title.scss
Comment (mirisuzanne) [inline src/scss/layout/_title.scss:56]: These should probably be specific to spans in `#title`? Unless we expect to use them elsewhere?

File: src/scss/layout/_title.scss
Comment (mirisuzanne) [inline src/scss/layout/_title.scss:58]: Why are we using a new font-family for this?

File: src/scss/layout/_title.scss
Comment (mirisuzanne) [inline src/scss/layout/_title.scss:60]: I don't think the mockup calls for a special font-weight? I think this could be `normal`.

File: src/scss/layout/_title.scss
Comment (mirisuzanne) [inline src/scss/layout/_title.scss:67]: Is this needed at all? Can we apply just the italics to the one word we want italicized?

**Review by SondraE — CHANGES_REQUESTED** [review]
It's so exciting to see this positioning update starting to come together!

**Intro Section:**
- Looks like the text "We Create" is in a different font instead of FreightSans Pro.

**Featured Posts**
- Can you give one of these posts the featured style so we can see how that looks?

**Full Range Section:**
- "As core contributors to CSS, Sass, and Django," Add comma after Sass in this sentence. 
- Move this section above Open Source Contributions.

I see you added the new link "View services & deliverables" to the Full Range section. Wanted to confirm that you're planning to add the new links to the Featured Clients, Featured Posts, and Open Source Contributions sections as well.

Thanks!

File: src/scss/layout/_title.scss
Comment (oluoluoxenfree) [reply src/scss/layout/_title.scss:58]: My mistake, for some reason I thought I checked the browser computed fonts and this came up? Must have been some other issue

File: content/_includes/post.macros.njk
Comment (oluoluoxenfree) [reply content/_includes/post.macros.njk:748]: @mirisuzanne `featured_home`? `homepage_posts`?

Comment (mirisuzanne) [reply content/_includes/post.macros.njk:748]: yeah, either one works for me

File: src/scss/layout/_title.scss
Comment (oluoluoxenfree) [reply src/scss/layout/_title.scss:13]: @mirisuzanne this is to bump the "we create" part of the heading past the top of the image next to it.

File: src/scss/layout/_title.scss
Comment (oluoluoxenfree) [reply src/scss/layout/_title.scss]: the `<em>` tag applies different styling to what i would expect; it currently makes the text look superscripted?

**Comment by oluoluoxenfree** [comment]
> **Featured Posts**
> 
> * Can you give one of these posts the featured style so we can see how that looks?

@SondraE can you remind me what a featured post is supposed to look like? I think you linked an image somewhere but i can't find it; is it the same as somewhere else on the site?

File: content/_includes/utility.macros.njk
Comment (jgerigmeyer) [reply content/_includes/utility.macros.njk:238]: It's sort of nice in this case to be able to have these on separate lines, but use the nunjucks `{-` syntax to remove line breaks. So I'm tempted to leave this?

File: content/index.njk
Comment (jgerigmeyer) [reply content/index.njk:67]: @mirisuzanne Yeah, speaking of nunjucks not having an active maintainer...

I ran into issues putting a custom tag (`{% md %}`) inside a `{% call %}` block. Turns out there's an [open issue](https://github.com/mozilla/nunjucks/issues/617)... from 2015! With a helpful "good luck with that" comment from our own @carljm, who was the maintainer at the time! 😂 

So I think we could look into `media_block` calling `md` internally, or via an option passed in -- but we can't use it within the `call` block directly in this file.

Comment (carljm) [reply content/index.njk:67]: Good luck with that! 😁

Comment (mirisuzanne) [reply content/index.njk:67]: eeep

File: src/scss/layout/_title.scss
Comment (mirisuzanne) [reply src/scss/layout/_title.scss]: oh interesting, I wonder where that's coming from. Have you looked at the CSS to see what's causing that?

Generally, we should avoid writing extra markup to work around unintentional styles - and instead try to understand and address where that style is coming from.

**Comment by mirisuzanne** [comment]
As far as I understand, the featured posts should look the same on the home page as in the resource lists. But rather than trying to recreate those styles, we really want to update the nunjucks logic so that they are getting the same markup in both places. Then the styles we want will be applied automatically.

File: src/scss/layout/_title.scss
Comment (mirisuzanne) [inline src/scss/layout/_title.scss:56]: I think this is the font family already used for headings? Is there a reason we need to specify it again?

File: content/index.njk
Comment (jgerigmeyer) [reply content/index.njk]: @mirisuzanne Adjusted in https://github.com/oddbird/oddleventy/commit/ba3561d839dd7ee280464c399c1d6a060afc5a30

**Review by jgerigmeyer — COMMENTED** [review]
This is design feedback for @SondraE rather than code review, but personally I feel like there's so much (too much) going on in this new intro:

Screenshot 2023-03-10 at 1 57 33 PM

It's a longer heading with more words, and in this one paragraph we've got so many different fonts/sizes/styles, it feels overwhelming to me. I'm not sure what the "We Create" is really adding? And I'm mixed on the "in Thoughtful Collaboration"... (Take this with a grain of salt, obviously -- I wanted to share my reaction, but happy to be out-voted by the Real Designers™ on the team!)

**Comment by SondraE** [comment]
@jgerigmeyer @mirisuzanne @stacyk 
[This is design feedback for @SondraE rather than code review, but personally I feel like there's so much (too much) going on in this new intro:](https://github.com/oddbird/oddleventy/pull/326#pullrequestreview-1335461530)

In our last conversation about this intro section we said:

- Keep tweaking as we implement
- Maybe "Thoughtful Process" instead of "Thoughtful Collaboration"
- See what "We Create" looks like in context and decide to keep or cut
- Think about whether there is a better term than "large-scale"
- Maybe remove "tailor-made"

I believe the time has come for "tweaking." What are your thoughts?

Adding my thoughts to Jonny's:

- I like highlighting our "Thoughtful Collaboration" in the header
- I'm leaning toward cutting "We Create."
- I'm not able to think of a better word than "large-scale." (Thesaurus.com suggests "prodigious.") ;)
- I think the sentence would read better if we removed "tailor-made," and I don't think we have to have it since we have "custom" in the headline above.
- I would also like to cut "with technology" from the final sentence in the paragraph. The grammar is awkward as it is now.

**Comment by SondraE** [comment]
@oluoluoxenfree 
Screenshot of the featured style:

**Comment by stacyk** [comment]
> @jgerigmeyer @mirisuzanne @stacyk @SondraE 
> 
> * I like highlighting our "Thoughtful Collaboration" in the header
> * I'm leaning toward cutting "We Create."
> * I'm not able to think of a better word than "large-scale." (Thesaurus.com suggests "prodigious.") ;)
> * I think the sentence would read better if we removed "tailor-made," and I don't think we have to have it since we have "custom" in the headline above.
> * I would also like to cut "with technology" from the final sentence in the paragraph. The grammar is awkward as it is now.


- The more I sit will this, the more "thoughtful collaboration" feels like jargon and the opening of the first sentence addresses how we work with others to ... and then it also states that "as thoughtful experts" again. IMO, I think we could remove this phrase from the large headline.
- I agree with the last sentence (a bit awkward sounding)
- Maintenance has a typo at the moment in the deploy preview. 
- we could remove some of the `<strong>` tags if it is too busy

Here's my suggestion is below:

We Create
Custom Web Apps
& Design Systems
We enjoy working with teams pursuing large-scale projects to plan, design, and develop web applications and design systems. As industry experts, we’d love to help you improve team workflow, lower maintenance costs, and thoughtfully solve problems using technology in order to help people.


home-intro

File: content/index.njk
Comment (jgerigmeyer) [reply content/index.njk]: @stacyk No idea; nunjucks/markdown is a riddle wrapped in a mystery inside an enigma. 🤷 

Addressed (poorly) in https://github.com/oddbird/oddleventy/commit/03a92c52569d7b727c4765e9ff5500f39d7cffcc

**Comment by stacyk** [comment]
I made the change to the grid for posts in lists (all posts in lists) and I'm not sure everyone will like it everywhere as opposed to showing it this way on homepage only. It isn't 'bad' on the posts page, but I do like the posts page better as is, and then having this new 1, 2 or 4up layout on homepage (or maybe where other posts are featured when there is a group of 2 or 4).

live-left-dev-right1

live-left-dev-right2

home-large
home-med
home-small

**Comment by mirisuzanne** [comment]
@stacyk I agree with your assessment of the two different card layouts

File: content/_includes/post.macros.njk
Comment (mirisuzanne) [inline content/_includes/post.macros.njk:145]: This is what the `attr_if()` utility macro does, if you want to use it. :)

File: content/_includes/post.macros.njk
Comment (jgerigmeyer) [inline content/_includes/post.macros.njk:201]: Adjusted the logic here so that we can set `override_feature` and `override_large` independently. If `override_large` is used without `override_feature`, `large` cards are converted to `feature` cards.

File: content/blog/2023/when-to-choose-a-responsive-web-app.md
Comment (jgerigmeyer) [inline content/blog/2023/when-to-choose-a-responsive-web-app.md:32]: Added this partially to test the automatic conversion of `large` cards (in the Posts list) down to `feature` cards (on the homepage). Feel free to change this if we don't want this post large or featured. (It looks like we're not using `card: feature` very frequently, at least recently?)

**Review by jgerigmeyer — APPROVED** [review]

**Review by SondraE — APPROVED** [review]
That looks good to me!

File: content/_includes/post.macros.njk
Comment (stacyk) [reply content/_includes/post.macros.njk:201]: Which means that anything marked as `card: large` will show up as large in the post lists, but it will look like `card: featured` on the homepage... Is that the intention of this?

File: content/blog/2023/when-to-choose-a-responsive-web-app.md
Comment (stacyk) [reply content/blog/2023/when-to-choose-a-responsive-web-app.md:32]: @jgerigmeyer ah, ok, this confirms my question from above. I wonder if we need a `card: feature` in the long run. I feel like having the three levels of posts is a little distracting but that is definitely not a part of this story.

**Review by stacyk — APPROVED** [review]
Looking forward to wrapping this one up soon. 🎉

File: content/blog/2023/when-to-choose-a-responsive-web-app.md
Comment (jgerigmeyer) [reply content/blog/2023/when-to-choose-a-responsive-web-app.md:32]: Sure -- I'd defer to you and @SondraE and @mirisuzanne on that, but agreed it's not part of this story.

File: content/_includes/post.macros.njk
Comment (jgerigmeyer) [reply content/_includes/post.macros.njk:201]: Correct!

**Pull request merged**
